.jenkins/jenkins.sh

```diff
@@ -22,21 +22,25 @@
 # JOB_URL            Full URL of this job, like http://server:port/jenkins/job/foo/
 
 # stop on all errors
-set +e
+set -e
 
 # get root directory of where jenkins.sh is sitting
 root=`dirname $0`
 envloc=`dirname $0`
 
 # some global variables
 action="$1"
-optarg="$2"
+shift
+optarg="$@"
```

Comment:
what is the reason for this change?

Reply:
Only the second argument was passed to the action, making something like `./jenkins action -compiler gnu` impossible since only `-compiler` would have been passed to `action.sh`. This now removes the `action` from the argument list using `shift` and then passes the remaining arguments to `action.sh`.